src/couch_replicator_ratelimiter.erl

```diff
@@ -0,0 +1,277 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+
+% This module implements rate limiting based on a variation the additive
+% increase / multiplicative decrease feedback control algorithm.
+%
+%  https://en.wikipedia.org/wiki/Additive_increase/multiplicative_decrease
+%
+% This is an adaptive algorithm which converges on available channel
+% capacity where each participant (client) doesn't a priori know the
+% capacity, and participants don't communicate or know about each other (so they
+% don't coordinate to divide the capacity among themselves).
+%
+% For convenience, estimated value is an interval (in milliseconds)
+% instead of a rate. Then the estimated rate = 1000 / interval. So technically
+% this implements "additive decrease / multiplicative increase", but it doesn't
+%  matter as long as the failure part is multiplicative and the success part is
+% additive. If that's the case the algorithm is proven to converge.
```

Comment:
This comment/paragraph here confused me a for a sec, so you may want to reword it a little.  Something along the lines of "The algorithm referenced above uses a rate, whereas the algorithm we implement uses an interval. It still implements  "additive decrease / multiplicative increase"... etc.